controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java

```diff
@@ -69,16 +71,67 @@ public boolean authenticate(String resource, String credentials, AuthHandler.Per
             String token = parts[1];
             AuthHandler handler = getHandler(method);
             assert handler != null;
-            if (!handler.authenticate(token)) {
+            Principal principal;
+            if ((principal = handler.authenticate(token)) == null) {
                 throw new AuthenticationException("Authentication failure");
             }
-            retVal = handler.authorize(resource, token).ordinal() >= level.ordinal();
+            retVal = handler.authorize(resource, principal).ordinal() >= level.ordinal();
         } catch (RuntimeException e) {
             throw new AuthenticationException(e);
+        } catch (AuthException e) {
+            throw new AuthenticationException("Authentication failure");
         }
         return retVal;
     }
 
+    /**
+     *
+     * API to authenticate a given credential.
+     * @param credentials  Credentials used for authentication.
+     *
+     * @return Returns the Principal if the entity represented by credentials is authenticated.
+     * @throws AuthException if an authentication failure occurred.
+     */
+    public Principal authenticate(String credentials) throws AuthException {
+        Preconditions.checkNotNull(credentials, "credentials");
```

Comment:
line 96-104 can be moved to a common convenience function used by authenticate and authorize

Reply:
oh right.  I guess lines 96-101 then.   The function would just return a String[].  (this is minor).

Also retVal doesn't seem to be used.

Reply:
Fixed. Refactored the code and dropped `retVal`.

Reply:
nit: retVal is still there